docs/source/miscellaneous/stubs.rst

```diff
@@ -468,27 +619,26 @@ An example from the built-in functions:
     */
    function get_declared_classes(): array {}
 
-Please note that the feature is only available for built-in stubs inside php-src, since currently
-there is no way to provide the function list for the optimizer other than overwriting
-``zend_func_infos.h`` directly.
+Functions can be evaluated at compile-time if their arguments are known in compile-time, and their
+behavior is free from side-effects and is not affected by the global state.
+
+The list of such functions in the optimizer was maintained manually until PHP 8.2.
 
-Additionally, functions can be evaluated at compile-time if their arguments are known in
-compile-time and their behavior if free from side-effects as well as it is not affected by the
-global state. Until PHP 8.2, a list of such functions was maintained manually in the optimizer.
-However, since PHP 8.2, the ``@compile-time-eval`` PHPDoc tag can be applied to any functions which
-conform to the above restrictions in order for them to qualify as evaluable at compile-time. The
-feature internally works by adding the ``ZEND_ACC_COMPILE_TIME_EVAL`` function flag.
+Since PHP 8.2, the ``@compile-time-eval`` PHPDoc tag can be applied to any function which conform to
+the above restrictions in order for them to qualify as evaluable at compile-time. The feature
+internally works by adding the ``ZEND_ACC_COMPILE_TIME_EVAL`` function flag.
 
-As of PHP 8.4, the concept of arity-based frameless functions was introduced. This is another
-optimization technique, which results in faster internal function calls by eliminating unnecessary
-checks for the number of passed parameters (if the number of passed arguments is known at
-compile-time).
+In PHP 8.4, arity-based frameless functions were introduced. This is another optimization technique,
+which results in faster internal function calls by eliminating unnecessary checks for the number of
+passed parameters—if the number of passed arguments is known at compile-time.
```

Comment:
spaces are missing around `-`?

Reply:
It seems controversial. I've always followed the Chicago Manual of Style (2.13) as much as I can (remember), which says "no spaces", but AP and newspapers usually do.

As long as it's consistent in this file (it is, as there is only one), it is fine.

Reply:
Ah ok, thanks for letting me know about this style. To be honest, I haven't heard about it yet.